### ISSUE TITLE
水平联邦学习的输入数据格式

### ISSUE THREAD

**Issue by LGDchampion** [secretflow/secretflow#963]
### Issue Type

Bug

### Source

binary

### Secretflow Version

SecretFlow version 1.2.0.dev20231009

### OS Platform and Distribution

Linux Ubuntu 20.04.6

### Python version

Python 3.8.10

### What happend and What you expected to happen.

```shell
我想尝试使用水平联邦学习功能，根据官方文档成功运行了《使用Pytorch后端来进行联邦学习》。
然后想将一般的pytorch框架模型使用secretflow改造成水平联邦学习的模式，但出现数据类型错误。比如下面这个例子：

原来torch框架：
我使用的模型是最简单的线性分类器，2维到2维，输入数据形式是（torch.tensor,形状(2,2)，dtype=torch.float32） 标签是（torch.tensor,形状(2),dtype=torch.long）,   交叉熵损失函数torch.nn.CrossEntropyLoss
这个使用torch是可以做出模型推断。

使用secretflow时的操作：
将输入数据变成np.ndarray: 
a= np.array([[1.1,1.2],[1.3,1.4]])
b= np.array([0,1],dtype=int)，
保存成.npz，然后用secretflow.data.ndarray.load读取，
按官方文档定义好TorchModel,FLModel,
调用FLModel.fit(),
然后报错：expected scalar type Long but found Float，    报错在torch的F.cross_entropy函数上。
```

### Reproduction code to reproduce the issue.

```shell
import torch
import numpy as np

import secretflow as sf

# Check the version of your SecretFlow
print('The version of SecretFlow: {}'.format(sf.__version__))

# In case you have a running secretflow runtime already.
sf.shutdown()

sf.init(['alice', 'bob', 'charlie'], address='local')
alice, bob, charlie = sf.PYU('alice'), sf.PYU('bob'), sf.PYU('charlie')


from secretflow.ml.nn.utils import BaseModule, TorchModel
from secretflow.ml.nn.fl.utils import metric_wrapper, optim_wrapper
from secretflow.ml.nn import FLModel
from torchmetrics import Accuracy, Precision
from secretflow.security.aggregation import SecureAggregator
from torch import nn, optim

#secretflow数据
a= np.array([[1.1,1.2],[1.3,1.4]])
b= np.array([0,1],dtype=int)

#pytorch数据
a_torch = torch.tensor(a,dtype=torch.float32)
b_torch = torch.tensor(b,dtype=torch.long)


#secretflow数据处理成fedndarray
np.savez("alice.npz", x=a, label=b)
np.savez("bob.npz", x=a, label=b)
from secretflow.data.ndarray import load
fed_npz = load(
    {alice: "./alice.npz", bob: "./bob.npz"}, allow_pickle=True, partition_way=sf.data.ndarray.ndarray.PartitionWay.HORIZONTAL
)
fed_image = fed_npz['x']
fed_label = fed_npz['label']


#secretflow模型
class Linear(BaseModule):
    def __init__(self):
        super(Linear, self).__init__()
        self.fc = torch.nn.Linear(2, 2)

    def forward(self, x):
        x = self.fc(x)
        return torch.nn.functional.softmax(x, dim=1)


#pytorch模型
class Linear_torch(torch.nn.Module):
    def __init__(self):
        super(Linear_torch,self).__init__()
        self.fc = torch.nn.Linear(2,2)

    def forward(self, x):
        x = self.fc(x)
        return torch.nn.functional.softmax(x, dim=1)

#使用pytorch进行推断
model = Linear_torch()
loss_f = torch.nn.CrossEntropyLoss()
z=model(a_torch)
print(z)
print(loss_f(z,b_torch))


#使用secretflow进行模型训练
loss_fn = nn.CrossEntropyLoss
optim_fn = optim_wrapper(optim.Adam, lr=1e-2)
model_def = TorchModel(
    model_fn=Linear,
    loss_fn=loss_fn,
    optim_fn=optim_fn,
    metrics=[
        metric_wrapper(Accuracy, task="multiclass", num_classes=10, average='micro'),
        metric_wrapper(Precision, task="multiclass", num_classes=10, average='micro'),
    ],
)

device_list = [alice, bob]
server = charlie
aggregator = SecureAggregator(server, [alice, bob])

# spcify params
fl_model = FLModel(
    server=server,
    device_list=device_list,
    model=model_def,
    aggregator=aggregator,
    strategy='fed_avg_w',  # fl strategy
    backend="torch",  # backend support ['tensorflow', 'torch']
)



history = fl_model.fit(
    fed_image,
    fed_label,
    validation_data=(fed_image, fed_label),
    epochs=1,
    batch_size=32,
    aggregate_freq=1,
)
```

**Comment by zhangxingmeng** [comment]
import torch
import numpy as np

import secretflow as sf

# Check the version of your SecretFlow
print('The version of SecretFlow: {}'.format(sf.__version__))

# In case you have a running secretflow runtime already.
sf.shutdown()

sf.init(['alice', 'bob', 'charlie'], address='local', debug_mode=True)
alice, bob, charlie = sf.PYU('alice'), sf.PYU('bob'), sf.PYU('charlie')


from secretflow.ml.nn.utils import BaseModule, TorchModel
from secretflow.ml.nn.fl.utils import metric_wrapper, optim_wrapper
from secretflow.ml.nn import FLModel
from torchmetrics import Accuracy, Precision
from secretflow.security.aggregation import SecureAggregator
from torch import nn, optim

# secretflow数据
a = np.array([[1.1, 1.2], [1.3, 1.4]])
b = np.array([[0, 1], [0, 1]], dtype=int)

# pytorch数据
a_torch = torch.tensor(a, dtype=torch.float32)
b_torch = torch.tensor(b, dtype=torch.long)


# secretflow数据处理成fedndarray
np.savez("alice.npz", x=a, label=b)
np.savez("bob.npz", x=a, label=b)
from secretflow.data.ndarray import load

fed_npz = load(
    {alice: "./alice.npz", bob: "./bob.npz"},
    allow_pickle=True,
    partition_way=sf.data.ndarray.ndarray.PartitionWay.HORIZONTAL,
)
fed_image = fed_npz['x']
fed_label = fed_npz['label']

fed_label = fed_label.astype(np.int64)


# secretflow模型
class Linear(BaseModule):
    def __init__(self):
        super(Linear, self).__init__()
        self.fc = torch.nn.Linear(2, 2)

    def forward(self, x):
        x = self.fc(x)
        return torch.nn.functional.softmax(x, dim=1)


# pytorch模型
class Linear_torch(torch.nn.Module):
    def __init__(self):
        super(Linear_torch, self).__init__()
        self.fc = torch.nn.Linear(2, 2)

    def forward(self, x):
        x = self.fc(x)
        return torch.nn.functional.softmax(x, dim=1)


# # 使用pytorch进行推断
# model = Linear_torch()
# loss_f = torch.nn.CrossEntropyLoss()
# z = model(a_torch)
# print(z)
# print(loss_f(z, b_torch))


# 使用secretflow进行模型训练
loss_fn = nn.CrossEntropyLoss
optim_fn = optim_wrapper(optim.Adam, lr=1e-2)
model_def = TorchModel(
    model_fn=Linear,
    loss_fn=loss_fn,
    optim_fn=optim_fn,
    metrics=[
        metric_wrapper(Accuracy, task="multiclass", num_classes=2, average='micro'),
        metric_wrapper(Precision, task="multiclass", num_classes=2, average='micro'),
    ],
)

device_list = [alice, bob]
server = charlie
aggregator = SecureAggregator(server, [alice, bob])

# spcify params
fl_model = FLModel(
    server=server,
    device_list=device_list,
    model=model_def,
    aggregator=aggregator,
    strategy='fed_avg_w',  # fl strategy
    backend="torch",  # backend support ['tensorflow', 'torch']
)


history = fl_model.fit(
    fed_image,
    fed_label,
    validation_data=(fed_image, fed_label),
    epochs=1,
    batch_size=32,
    aggregate_freq=1,
)
